### DISCUSSION TITLE
Roadmap towards package announcement paper

### DISCUSSION THREAD

**Discussion by navidcy** [COSIMA/mom6-regional-scripts#42] [Ideas]
Thought I start this to list few things that need to be in place before we draft an announcement paper of the package and submit to some open-source software journal.

There are a few candidate journals we can go for; [JOSS](https://joss.theoj.org) is one, [JORS](https://openresearchsoftware.metajnl.com) is another one. Personally, I've had [experience](https://doi.org/10.21105/joss.03053) with the former and it was a very [nice experience](https://climateextremes.org.au/developing-and-publishing-open-source-software-and-my-experience-with-an-open-peer-review-process/).

- [x] convert on package name; see #33 
- [x] testing CI pipeline
- [ ] meaningful tests and achieve as high code coverage as we can (e.g. we should aim for >75%)
- [ ] documentation (basic description, basic installation instructions #40, API list w curated docstrings, example(s))
- [ ] demo notebook + incorporate this in the documentation
- [ ] clear up all references to particular dataset paths; convert methods to allow give a directory or array or xarray dataset as input
- [ ] clear up all references towards gadi paths from the source code
- [ ] create pipelines for automated register and push to conda and/or pypi pipelines as soon as we create a new release via Github

**Comment by navidcy** [comment]
(cc @angus-g, @ashjbarnes, @AndyHoggANU, @aekiss, @adele157; comments welcome)

**Reply by ashjbarnes** [reply]
A big one is including some more options for vertical and horizontal grid generation. Currently there are two placeholders which happen to be the algorithms I'm using for my personal research. More options should be included, probably leveraging the `Grid Gen Tools` package. See issue #69

**Comment by navidcy** [comment]
This recent [paper](https://doi.org/10.5194/gmd-16-1481-2023) is definitely relevant.
